### DISCUSSION TITLE
How can I draw only one pixel ?

### DISCUSSION THREAD

**Discussion by seleniakos** [emilk/egui#2984] [Q&A]
Hello, new in Rust and egui world, I was in vain looking for an example to draw one single pixel, something like "painter.draw_pixel(x, y, color)"... Can anybody give me a hint ?

**Comment by pm100** [comment]
did you ever find out

**Comment by 9SMTM6** [comment]
Canvas allow you to draw with single pixel precision. Usually you use that to draw multiple pixels but nobody is stopping you. The demo app has an example with glow. 

But doing that stuff is fairly advanced.
IDK if theres easier options.

**Comment by pm100** [comment]
Heres how I did it

```
        let draw_area_size =
            Vec2::new(SCREEN_WIDTH as f32, SCREEN_HEIGHT as f32) + Vec2::splat(1.0);
        let (response, painter) = ui.allocate_painter(draw_area_size, Sense::hover());
```

then 

```
                    let rect_points =
                                egui::Rect::from_min_size(Pos2::new(x, y), Vec2::splat(1.0));
                            painter.rect_filled(rect_points, 0.0, color);

```

so get a painter from any Ui and then draw a 1 by 1 rectangle
